pkg/webhooks/resource/generation/handler.go

```diff
@@ -115,7 +115,7 @@ func (h *generationHandler) handleTrigger(
 		for _, rule := range engineResponse.PolicyResponse.Rules {
 			if rule.Status() == engineapi.RuleStatusPass {
 				appliedRules = append(appliedRules, rule)
-			} else if rule.Status() == engineapi.RuleStatusFail {
+			} else if rule.Status() == engineapi.RuleStatusFail || rule.Status() == engineapi.RuleStatusSkip {
```

Comment:
Thanks for the PR @mikebryant ! 

Instead of considering rule status skip, would it make sense to return status=fail if the `preconditions` of the new resource no longer matches while the old does?

Otherwise we will apply skipped rules even if they don't match in the first place, which seems unnecessary to me.

Reply:
>I'm not sure it would make sense - the current status are defined clearly here: https://github.com/kyverno/kyverno/blob/main/pkg/engine/api/rulestatus.go#L18-L20

That's true. But here the case is a bit different, the clean-up happens when the new resource no longer matches the rule while the old still matches. If we go with this change, Kyverno will create the UR on all skipped rules, which brings unnecessary extra events to the cluster.

Reply:
@mikebryant - what's your thought on this?

Reply:
Any other follow up?